driver-core/src/main/com/mongodb/client/model/search/SearchOperators.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.search;
+
+import org.bson.BsonBoolean;
+import org.bson.BsonObjectId;
+import org.bson.types.ObjectId;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+
+/**
+ * Builder API for Atlas search.
+ *
+ * @since 4.4
+ * @mongodb.driver.manual.atlas/reference/atlas-search/query-syntax/
+ */
+public interface SearchOperators {
```

Comment:
The main entry point of the API.

Reply:
I'm still not used to it, feels a miss use of an interface as its not designed to be implemented.

Reply:
I can't find the JEP for this feature, but I think that this use case was one of the motivations.

Reply:
:+1:

Reply:
> improvement over a final class in that it expresses intent more clearly

The intent is to have a place with a bunch of `static` methods. A `final` class with no constructor clearly expresses this intent, while an interface does not, because an interface defines a type and expresses an intent to for this type to be implemented.

I had a similar conversation with @rozza here https://github.com/mongodb/mongo-spark/pull/54#discussion_r716296848, and instead of copying it, I am linking it.